modules/iknowpy/tests/udct.py

```diff
@@ -26,20 +49,40 @@
 user_dictionary.add_number("magic number")
 user_dictionary.add_time("future")
 
+if len(user_dictionary.entries) != 11:
+    print("ERROR: UD not fully loaded!")
+
 ret = engine.load_user_dictionary(user_dictionary)
-engine.index("some text Fr. w/o one concept and crap one relation that's great and awfull, magic number 3 Hg from future", "en", True) # generate Traces
-for trace in engine.m_traces:
-    key, value = trace.split(':', 1)[0],trace.split(':', 1)[1]
-    if (key=='UserDictionaryMatch'):
-        print(value)
+engine.index(test_sentence, "en", True) # generate Traces
+count_matches(engine.m_traces, 'UserDictionaryMatch', 11, debug)
 
+print("\nUnloading User Dictionary...")
 engine.unload_user_dictionary()
-engine.index("some text Fr. w/o one concept and crap one relation that's great and awfull, magic number 3 Hg from future", "en", True) # generate Traces
-for trace in engine.m_traces:
-    key, value = trace.split(':', 1)[0],trace.split(':', 1)[1]
-    if (key=='LexrepIdentified'):
-        print(value)
+engine.index(test_sentence, "en", True) # generate Traces
+count_matches(engine.m_traces, 'LexrepIdentified', -1, debug)
+count_matches(engine.m_traces, 'UserDictionaryMatch', 0)
+
+# test array load
+print("\nBuilding User Dictionary from list...")
+user_dictionary_2 = iknowpy.UserDictionary(user_dictionary.entries)
+if len(user_dictionary_2.entries) != 11:
+    print("ERROR: UD not fully loaded!")
+engine.load_user_dictionary(user_dictionary_2)
+engine.index(test_sentence, "en", True) # generate Traces
+count_matches(engine.m_traces, 'UserDictionaryMatch', 11)
 
+
+# test clear
+print("\nClearing User Dictionary ...")
+user_dictionary_2.clear()
+if len(user_dictionary_2.entries) != 0:
+    print("ERROR: UD not fully cleared!")
+engine.unload_user_dictionary() # why is this needed?
+engine.load_user_dictionary(user_dictionary_2)
```

Comment:
@JosDenysGitHub can you take a look at the `clear()` method of the UserDictionary() object (C++ side). It doesn't look like it's actually clearing out. If you leave this line uncommented, you'll see that the engine still gets the old UD contents.